### PR TITLE
Initial v3 project setup

### PR DESCRIPTION
**Pull request by yufeih**
- Basic documentation: readme, contribution, developer guide
- Basic build and test script
- Setup stylecop and fxcop

#2534

### CODE REVIEW

File: docs/engineering/developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Developer Guide
+
+> __v3__ develop work currently happens in _v3.0_ branch, these instructions only applies to source code in _v3.0_ branch.
```
Comment (vicancy) [inline docs/engineering/developer-guide.md]: >v3.0 [](start = 44, length = 4)

v3

File: README.md [hunk 2]
```diff
@@ -1,86 +1,25 @@
...
 
-|            | Build Status  |  Package   |  Chocolatey |
-| ---------- | ------------- | ---------- | ----------- |
-| **master** |[![masterstatus](https://img.shields.io/teamcity/http/docfx-ci-0.cloudapp.net/s/DocfxCiWithScripts_DocfxCiForMasterBranch.svg?label=master)](http://docfx-ci-0.cloudapp.net/viewType.html?buildTypeId=DocfxCiWithScripts_DocfxCiForMasterBranch) |[![NuGet](https://img.shields.io/nuget/v/docfx.svg)](http://www.nuget.org/packages/docfx/) |[![Chocolatey](https://img.shields.io/chocolatey/v/docfx.svg)](https://chocolatey.org/packages/docfx)
-|  **dev**   |[![devstatus](https://img.shields.io/teamcity/http/docfx-ci-0.cloudapp.net/s/DocfxCiWithScripts_DocfxCiForDevBranch.svg?label=dev)](http://docfx-ci-0.cloudapp.net/viewType.html?buildTypeId=DocfxCiWithScripts_DocfxCiForDevBranch) |[![MyGet](https://img.shields.io/myget/docfx-dev/v/docfx.svg?label=myget)](https://www.myget.org/feed/Packages/docfx-dev)
+> We are still at an early develop phase, see our [roadmap](docs/roadmap.md) for detailed planing.
 
-* Running Status
+## Contributing
```
Comment (vicancy) [inline README.md]: >Contributing [](start = 3, length = 12)

use noun?

File: stylecop.ruleset [hunk 3]
```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="Microsoft Managed Recommended Rules" Description="These rules focus on the most critical problems in your code, including potential security holes, application crashes, and other important logic and design errors. It is recommended to include this rule set in any custom rule set you create for your projects." ToolsVersion="10.0">
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
+    <!-- SA0001XmlCommentAnalysisDisabled -->
+    <Rule Id="SA0001" Action="None" />
+    <!-- SA1101PrefixLocalCallsWithThis -->
+    <Rule Id="SA1101" Action="None" />
+    <!-- SA1122UseStringEmptyForEmptyStrings -->
+    <Rule Id="SA1122" Action="None" />
```
Comment (vicancy) [inline stylecop.ruleset]: >None [](start = 30, length = 4)

Warning

File: stylecop.ruleset [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    <!-- SA1127GenericTypeConstraintsMustBeOnOwnLine -->
+    <Rule Id="SA1127" Action="None" />
+    <!-- SA1308VariableNamesMustNotBePrefixed -->
+    <Rule Id="SA1308" Action="None" />
+    <!-- SA1309FieldNamesMustNotBeginWithUnderscore -->
+    <Rule Id="SA1309" Action="None" />
+    <!-- SA1311StaticReadonlyFieldsMustBeginWithUpperCaseLetter -->
+    <Rule Id="SA1311" Action="None" />
+    <!-- SA1401FieldsMustBePrivate -->
+    <Rule Id="SA1401" Action="None" />
```
Comment (vicancy) [inline stylecop.ruleset]: >None [](start = 30, length = 4)

why?

File: stylecop.ruleset [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    <!-- SA1402FileMayOnlyContainASingleType -->
+    <Rule Id="SA1402" Action="None" />
```
Comment (vicancy) [inline stylecop.ruleset]: >None [](start = 30, length = 4)

How about enabling it? It makes world easy when using GitHub `Find Files` to quickly locate to the class

File: stylecop.ruleset [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    <!-- SA1405DebugAssertMustProvideMessageText -->
+    <Rule Id="SA1405" Action="None" />
+    <!-- SA1502ElementMustNotBeOnSingleLine -->
+    <Rule Id="SA1502" Action="None" />
+    <!-- SA1503BracesMustNotBeOmitted -->
+    <Rule Id="SA1503" Action="None" />
```
Comment (vicancy) [inline stylecop.ruleset]: >None [](start = 30, length = 4)

Enable?

Comment (yufeih) [reply stylecop.ruleset]: Per our style guide, braces can be omitted in some cases:
> A single line statement block can go without braces but the block must be properly indented on its own line and it must not be nested in other statement blocks that use braces

File: README.md (hunk 2 above)
Comment (yufeih) [reply README.md]: Same as CONTRIBUTING.md

File: README.md [hunk 7]
```diff
@@ -1,86 +1,27 @@
-# Generate your API documentation with *DocFX*
+# Build your docs with DocFX
 
+[![Windows Build Status](https://ci.appveyor.com/api/projects/status/pqqfyw0ag3w4p0es/branch/v3?svg=true)](https://ci.appveyor.com/project/yufeih/docfx)
```
Comment (superyyrrzz) [inline README.md]: >https://ci.appveyor.com/project/yufeih/docfx [](start = 107, length = 44)

move to a APEX team account?

File: stylecop.ruleset [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+    <!-- SA1122UseStringEmptyForEmptyStrings -->
+    <Rule Id="SA1122" Action="None" />
+    <!-- SA1127GenericTypeConstraintsMustBeOnOwnLine -->
+    <Rule Id="SA1127" Action="None" />
+    <!-- SA1308VariableNamesMustNotBePrefixed -->
+    <Rule Id="SA1308" Action="None" />
+    <!-- SA1309FieldNamesMustNotBeginWithUnderscore -->
+    <Rule Id="SA1309" Action="None" />
+    <!-- SA1311StaticReadonlyFieldsMustBeginWithUpperCaseLetter -->
+    <Rule Id="SA1311" Action="None" />
```
Comment (superyyrrzz) [inline stylecop.ruleset]: >None [](start = 30, length = 4)

enable?

Comment (yufeih) [reply stylecop.ruleset]: This ruleset is based on https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

File: stylecop.ruleset (hunk 3 above)
Comment (yufeih) [reply stylecop.ruleset]: Lets use ""

**Pull request merged**